### PR TITLE
[TS] InifiniteScrollProps TS handler improve #3165

### PR DESCRIPTION
**Pull request by fredhase**
#### What does this PR do?
Fixes the types in src/js/components/InfiniteScroll/index.d.ts

#### Where should the reviewer start?
src/js/components/InfiniteScroll/

#### What testing has been done on this PR?
Tested using the test that already exists in the component and using 'yarn storybook'.

#### How should this be manually tested?
Testing InfiniteScroll

#### What are the relevant issues?
#3165 

#### Do the grommet docs need to be updated?
I believe not

#### Should this PR be mentioned in the release notes?
I believe not

#### Is this change backwards compatible or is it a breaking change?
I believe it's compatible

### CODE REVIEW

**Comment by ShimiSun** [comment]
Thank you for helping us improve typescript! 
We added Storybook for typescript, so any js story file could be rewritten as tsx file and could be verified with the typescript build process.
I copy/pasted the [onMore story](https://storybook.grommet.io/?path=/story/infinitescroll--onmore) and transformed it to typescript, but it failed with your suggested changes of `children`.

Feel free to add a typescript example file for your changes to verify and test your suggestions for any prop you are changing.
This is the file I created that fails to pass, you can use it as a reference.
```
import React, { useState } from 'react';
import { storiesOf } from '@storybook/react';

import {
  Grommet,
  Box,
  InfiniteScroll,
  InfiniteScrollProps,
  Text,
} from 'grommet';
import { grommet } from 'grommet/themes';

import { allItems } from '../Basics';

export interface IProps {
  props?: InfiniteScrollProps;
  step?: InfiniteScrollProps['step'];
}

const OnMoreInfiniteScroll: React.FC<IProps> = ({ props }) => {
  const [items, setItems] = useState(allItems.slice(0, 50));

  const onMore = () => {
    setTimeout(() => {
      setItems(allItems.slice(0, items.length + 50));
    }, 1000);
  };

  return (
    <Grommet theme={grommet}>
      <Box>
        <InfiniteScroll items={items} onMore={onMore} {...props}>
          {item => (
            <Box
              key={item}
              pad="medium"
              border={{ side: 'bottom' }}
              align="center"
            >
              <Text>{item}</Text>
            </Box>
          )}
        </InfiniteScroll>
      </Box>
    </Grommet>
  );
};

storiesOf('InfiniteScroll', module)
  .add('onMore', () => <OnMoreInfiniteScroll />)
  .add('onMore step', () => <OnMoreInfiniteScroll step={75} />);

```

File: src/js/components/InfiniteScroll/index.d.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import * as React from "react";
 
 export interface InfiniteScrollProps {
-  children?: ((...args: any[]) => any);
-  items?: any[];
-  onMore?: ((...args: any[]) => any);
-  renderMarker?: ((...args: any[]) => any);
+  children?: ((...args: [string, number] ) => React.ElementType);
```
Comment (ShimiSun) [inline src/js/components/InfiniteScroll/index.d.ts]: This failed the test I attached.

Comment (fredhase) [reply src/js/components/InfiniteScroll/index.d.ts]: Commited a change that made everything ok with the file you attached :)
Everythings seens to bem running fine with the types. Many thanks for the help

**Comment by ShimiSun** [comment]
Hi @fredhase , please add the attached file to your PR, and please add to the file example of the props you are changing, so we can be sure your changes are backward compatible.

**Comment by fredhase** [comment]
@ShimiSun The tsx is now in the new commit :)

File: src/js/components/InfiniteScroll/index.d.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 
 export interface InfiniteScrollProps {
-  children?: ((...args: any[]) => any);
-  items?: any[];
-  onMore?: ((...args: any[]) => any);
-  renderMarker?: ((...args: any[]) => any);
+  children?: ((...args: [string, number] ) => React.View);
+  items?: string[];
+  onMore?: (() => void);
+  renderMarker?: ((...args: React.ElementType[]) => React.ElementType);
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/js/components/InfiniteScroll/index.d.ts:7]: Items could also be an array of objects. For example:
```
const items = [
 {name: 'Joe', age: 22},
 {name: 'Stacy', age: 30},
 {name: 'Ricardo', age: 16},
 {name: 'Bob', age: 25}
]
```

**Pull request closed without merging**